ballerina/sse_to_bytes_stream_generator.bal

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2024 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/log;
+
+const LINE_BREAK = "\n";
+
+class SseEventToByteStreamGenerator {
+    private final stream<SseEvent, error?>|stream<SseEvent, error> eventStream;
+    private boolean isClosed = false;
+    private boolean isErrorOccurred = false;
+
+    isolated function init(stream<SseEvent, error?> eventStream) {
+        self.eventStream = eventStream;
+    }
+
+    public isolated function next() returns record {|byte[] value;|}|io:Error? {
+        if self.isClosed || self. isErrorOccurred {
+            return ();
+        }
+        do {
+            record {SseEvent value;}? event = check self.eventStream.next();
+            if event is () {
+                check self.close();
+                return ();
+            }
+            check validateSseEvent(event.value);
+            string eventText = getEventText(event.value);
+            return {value: eventText.toBytes()};
+        } on fail error e {
+            self.isErrorOccurred = true;
+            log:printError("Unable to obtain byte array", e);
```

Comment:
Do we need to log this error?

Reply:
Yes, this decision was made during the code review call.